<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Quiroz">

<title>Microbiome Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="analysis_files/libs/quarto-html/quarto.js"></script>
<script src="analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Microbiome Data Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Quiroz </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>Analysis pipeline base on this repository: <a href="https://gitlab.com/DanielQuiroz97/bacteria_ark/-/blob/master/CAYAMBE/Year%201/Analisis_final.Rmd">Bacteria Ark</a></p>
<section id="importing-libraries" class="level1">
<h1>Importing Libraries</h1>
<div class="cell" data-paged.print="false">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ranacapa)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(psadd)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="importing-data" class="level1">
<h1>Importing data</h1>
<section id="importing-raw-data" class="level2">
<h2 class="anchored" data-anchor-id="importing-raw-data">Importing raw data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tax table</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>tax_file <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"Data/asv-taxonomy-mapping.tsv"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ASV table</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>asv_count <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"Data/asvtable-processed-absolute.tsv"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>micro_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"Data/sample-metadata.csv"</span>, <span class="at">stringsAsFactors =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="modifying-tables-fro-phyloseq" class="level2">
<h2 class="anchored" data-anchor-id="modifying-tables-fro-phyloseq">Modifying tables fro phyloseq</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ASV table</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>asv_names <span class="ot">&lt;-</span> asv_count<span class="sc">$</span>asv</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>asv_count <span class="ot">&lt;-</span> asv_count <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>asv) </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(asv_count) <span class="ot">&lt;-</span> asv_names </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Setting row names on a tibble is deprecated.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(<span class="fu">as.matrix</span>(asv_count), <span class="at">taxa_are_rows =</span> T)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>micro_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(micro_metadata)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(micro_metadata) <span class="ot">&lt;-</span> micro_metadata<span class="sc">$</span>NAME</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>SAMPLE <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(micro_metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="separating-tax-single-column-into-multiple-column" class="level2">
<h2 class="anchored" data-anchor-id="separating-tax-single-column-into-multiple-column">Separating Tax single column into multiple column</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tax cleaning</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://www.yanh.org/2021/01/01/microbiome-r/</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># clean the taxonomy, Greengenes format</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> tax_file <span class="sc">%&gt;%</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(tax) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(tax, <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(str_sub, <span class="at">start =</span> <span class="dv">4</span>, <span class="at">end =</span> <span class="sc">-</span><span class="dv">4</span>) </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">replace</span>(tax, tax <span class="sc">==</span> <span class="st">""</span>, <span class="st">"Unclassified"</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tax) <span class="ot">&lt;-</span> tax_file<span class="sc">$</span>asv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Setting row names on a tibble is deprecated.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>TAX <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(<span class="fu">as.matrix</span>(tax))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="creating-a-phyloseq-object" class="level2">
<h2 class="anchored" data-anchor-id="creating-a-phyloseq-object">Creating a phyloseq object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>floral_microbiome_raw <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(ASV, TAX, SAMPLE)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>floral_microbiome_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4256 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 4256 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<ul>
<li>4256 taxa</li>
<li>35 samples</li>
<li>29 variables in metadata</li>
</ul>
</section>
</section>
<section id="filtering-data" class="level1">
<h1>Filtering data</h1>
<section id="removing-unclassified" class="level2">
<h2 class="anchored" data-anchor-id="removing-unclassified">Removing Unclassified</h2>
<section id="unclassified-at-kindom-level-and-archea-asv" class="level3">
<h3 class="anchored" data-anchor-id="unclassified-at-kindom-level-and-archea-asv">Unclassified at Kindom level and Archea ASV</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>floral_micro_filter <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(floral_microbiome_raw,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                                   Kingdom <span class="sc">!=</span> <span class="st">"Unclassified"</span> <span class="sc">&amp;</span> Kingdom <span class="sc">!=</span> <span class="st">"Archaea"</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>floral_micro_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4253 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 4253 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<p>We remove 3 taxa that are either <em>Unclassified</em> or <em>Archaea</em></p>
</section>
<section id="removing-cloroplast" class="level3">
<h3 class="anchored" data-anchor-id="removing-cloroplast">Removing cloroplast</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>floral_micro_filter <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(floral_micro_filter,</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                                   Order <span class="sc">!=</span> <span class="st">"Chloropl"</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>floral_micro_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4197 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 4197 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<p>We removed 53 ASV assigned as <em>Chloroplast</em> at the Order level</p>
</section>
<section id="unclassified-at-phylum-level" class="level3">
<h3 class="anchored" data-anchor-id="unclassified-at-phylum-level">Unclassified at Phylum level</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>floral_micro_filter <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(floral_micro_filter,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                                   Phylum <span class="sc">!=</span> <span class="st">"Unclassified"</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>floral_micro_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4150 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 4150 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="basis-read-info" class="level1">
<h1>Basis Read info</h1>
<section id="sequencing-depth" class="level2">
<h2 class="anchored" data-anchor-id="sequencing-depth">Sequencing depth</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sequencing depth</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>SeqDepth <span class="ot">=</span> <span class="fu">colSums</span>(<span class="fu">otu_table</span>(floral_micro_filter))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add depth to the metadata</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(floral_micro_filter)<span class="sc">$</span>SeqDepth <span class="ot">&lt;-</span> SeqDepth</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="fu">meta</span>(floral_micro_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(NAME, SeqDepth,<span class="at">fill =</span> <span class="fu">factor</span>(Treatment))) <span class="sc">+</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">scale_fill_cosmic</span>() <span class="sc">+</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>() <span class="sc">+</span> </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Treatment</span><span class="sc">\n</span><span class="st">Fertilizer (ppm)"</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Sample Name"</span>, <span class="at">y =</span> <span class="st">"Secuencing Depth"</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Secuending depth per sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="removing-singletons" class="level2">
<h2 class="anchored" data-anchor-id="removing-singletons">Removing singletons</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#checking for singletons</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">taxa_sums</span>(floral_micro_filter) <span class="sc">==</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6</code></pre>
</div>
</div>
<p>6 ASV with only 1 count, these are considered chimeric and therefore, are removed</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>floral_micro_filter <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(floral_micro_filter) <span class="sc">&gt;</span> <span class="dv">5</span>,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                                  floral_micro_filter)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>floral_micro_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2661 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 30 sample variables ]
tax_table()   Taxonomy Table:    [ 2661 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
</section>
<section id="kronas-plot" class="level2">
<h2 class="anchored" data-anchor-id="kronas-plot">Kronas plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Kronas per sample</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_krona</span>(<span class="at">physeq =</span> floral_micro_filter,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">output =</span> <span class="st">"Figures/Kronas/per_sample"</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">variable =</span> <span class="st">"NAME"</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_krona</span>(<span class="at">physeq =</span> floral_micro_filter,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">output =</span> <span class="st">"Figures/Kronas/per_treatment"</span>,</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">variable =</span> <span class="st">"Treatment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rarefraction-curves" class="level2">
<h2 class="anchored" data-anchor-id="rarefraction-curves">Rarefraction curves</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2023</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>plt_rarefraction <span class="ot">&lt;-</span> <span class="fu">ggrare</span>(floral_micro_filter, <span class="at">step =</span> <span class="dv">1000</span>, </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">label =</span> <span class="st">"NAME"</span>, <span class="at">se =</span> F) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Rarefraction curves per sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>rarefying sample L1
rarefying sample L10
rarefying sample L11
rarefying sample L12
rarefying sample L13
rarefying sample L14
rarefying sample L15
rarefying sample L16
rarefying sample L17
rarefying sample L18
rarefying sample L19
rarefying sample L2
rarefying sample L20
rarefying sample L21
rarefying sample L22
rarefying sample L23
rarefying sample L24
rarefying sample L25
rarefying sample L26
rarefying sample L27
rarefying sample L28
rarefying sample L29
rarefying sample L3
rarefying sample L30
rarefying sample L31
rarefying sample L32
rarefying sample L33
rarefying sample L34
rarefying sample L35
rarefying sample L36
rarefying sample L4
rarefying sample L6
rarefying sample L7
rarefying sample L8
rarefying sample L9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="relative-barchar" class="level2">
<h2 class="anchored" data-anchor-id="relative-barchar">Relative Barchar</h2>
<section id="phyla-level" class="level3">
<h3 class="anchored" data-anchor-id="phyla-level">Phyla level</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate relative abundances</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>floral_micro_rel <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(floral_micro_filter,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate taxa genus </span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>floral_glom_fl <span class="ot">&lt;-</span> <span class="fu">tax_glom</span>(floral_micro_rel, <span class="at">taxrank =</span> <span class="st">"Family"</span>, <span class="at">NArm =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>floral_glom_melt_fl <span class="ot">&lt;-</span> <span class="fu">psmelt</span>(floral_glom_fl)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>floral_glom_melt_fl<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">as.character</span>(floral_glom_melt_fl<span class="sc">$</span>Family)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating </span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>floral_gm_sum_fl <span class="ot">&lt;-</span> floral_glom_melt_fl <span class="sc">%&gt;%</span> </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(NAME, Family, Treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Abundance =</span> <span class="fu">sum</span>(Abundance))</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>floral_gm_sum_fl <span class="ot">&lt;-</span> floral_gm_sum_fl <span class="sc">%&gt;%</span> </span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Treatment =</span> <span class="fu">factor</span>(Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">100</span>, <span class="dv">200</span>) )) <span class="sc">%&gt;%</span> </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> <span class="fu">case_when</span>(Abundance <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">~</span> <span class="st">"&lt; 5%"</span>,</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> Family) )</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(floral_gm_sum_fl, <span class="fu">aes</span>(NAME, Abundance, <span class="at">fill =</span> Family) ) <span class="sc">+</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="st">"Treatment"</span>, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">rotate_x_text</span>() <span class="sc">+</span> <span class="fu">scale_fill_aaas</span>()  <span class="sc">+</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample Name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="genus-level" class="level3">
<h3 class="anchored" data-anchor-id="genus-level">Genus level</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate relative abundances</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate taxa genus </span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>floral_glom <span class="ot">&lt;-</span> <span class="fu">tax_glom</span>(floral_micro_rel, <span class="at">taxrank =</span> <span class="st">"Genus"</span>, <span class="at">NArm =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>floral_glom_melt <span class="ot">&lt;-</span> <span class="fu">psmelt</span>(floral_glom)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>floral_glom_melt<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">as.character</span>(floral_glom_melt<span class="sc">$</span>Genus)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating </span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>floral_gm_sum <span class="ot">&lt;-</span> floral_glom_melt <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(NAME, Genus, Treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Abundance =</span> <span class="fu">sum</span>(Abundance))</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>floral_gm_sum <span class="ot">&lt;-</span> floral_gm_sum <span class="sc">%&gt;%</span> </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Treatment =</span> <span class="fu">factor</span>(Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">100</span>, <span class="dv">200</span>) )) <span class="sc">%&gt;%</span> </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">case_when</span>(Abundance <span class="sc">&lt;</span> <span class="dv">3</span><span class="sc">~</span> <span class="st">"&lt; 3%"</span>,</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> Genus) )</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(floral_gm_sum, <span class="fu">aes</span>(NAME, Abundance, <span class="at">fill =</span> Genus) ) <span class="sc">+</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="st">"Treatment"</span>, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">rotate_x_text</span>() <span class="sc">+</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample Name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>
<section id="diversity-calculations" class="level1">
<h1>Diversity calculations</h1>
<section id="alpha-diversity" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity">Alpha diversity</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(floral_micro_filter)<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">factor</span>(<span class="fu">sample_data</span>(floral_micro_filter)<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">100</span>, <span class="dv">200</span>))</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>richness <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(floral_micro_filter, </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Chao1"</span>, <span class="st">"Simpson"</span>, <span class="st">"Shannon"</span>))</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>plt_rich_chao1 <span class="ot">&lt;-</span> <span class="fu">plot_richness</span>(floral_micro_filter, </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Chao1"</span>), <span class="at">nrow =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_aaas</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">rotate_x_text</span>() <span class="sc">+</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(variable<span class="sc">~</span>Treatment )</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>plt_rich_chao1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="deseq2-analysis" class="level1">
<h1>Deseq2 analysis</h1>
<section id="remove-poor-represented-asv" class="level2">
<h2 class="anchored" data-anchor-id="remove-poor-represented-asv">Remove poor represented ASV</h2>
<p>In this case, since we are going to conduct comparative analysis, we need to remove ASV with less that 5 counts as well as to be detected in at least 50% of the samples. This yields to a total of 534 taxa that meet this two criteria.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>floral_micro_filter_5C <span class="ot">&lt;-</span> <span class="fu">genefilter_sample</span>(floral_micro_filter,</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">filterfun_sample</span>(<span class="cf">function</span>(x) x <span class="sc">&gt;</span> <span class="dv">5</span>),</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">A=</span><span class="fl">0.5</span><span class="sc">*</span><span class="fu">nsamples</span>(floral_micro_filter)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>floral_micro_deseq <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(floral_micro_filter_5C, floral_micro_filter)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>floral_micro_deseq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 534 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 30 sample variables ]
tax_table()   Taxonomy Table:    [ 534 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
</section>
<section id="transform-counts-to-a-even-depth" class="level2">
<h2 class="anchored" data-anchor-id="transform-counts-to-a-even-depth">Transform counts to a even depth</h2>
<p>As we can see, 71080 is an intermediate sequencing depth for treatment. We are going to transform the count table from absolute counts to an homogeneous depth.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meta</span>(floral_micro_deseq) <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Depth_mean =</span> <span class="fu">mean</span>(SeqDepth))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 Ã— 2
  Treatment Depth_mean
  &lt;fct&gt;          &lt;dbl&gt;
1 25            64103.
2 100           71081.
3 200           71230.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>floral_micro_relat <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(floral_micro_deseq,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>                                              <span class="cf">function</span>(x) <span class="dv">71080</span> <span class="sc">*</span> x<span class="sc">/</span>(<span class="fu">sum</span>(x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="keeping-the-6-most-abundant-phyla" class="level2">
<h2 class="anchored" data-anchor-id="keeping-the-6-most-abundant-phyla">Keeping the 6 most abundant phyla</h2>
<p>Only the six most abundant phyla are preserved for downstream analysis to remove low represented ASV.</p>
<p>This can be view in the Kronas plot and the Top Six Phylum are</p>
<ul>
<li><p>Proteobacte</p></li>
<li><p>Actinobacteri</p></li>
<li><p>Bacteroid</p></li>
<li><p>Verrucomicrobi</p></li>
<li><p>Plantcomycet</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>phylum_sum <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">taxa_sums</span>(floral_micro_relat),</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">tax_table</span>(floral_micro_relat)[, <span class="st">"Phylum"</span>], sum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>top5phyla <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(phylum_sum, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="fu">seq</span>(<span class="dv">6</span>)]</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>top5phyla</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Proteobacte"    "Actinobacteri"  "Bacteroid"      "Verrucomicrobi"
[5] "Planctomycet"   "Acidobacteri"  </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>floral_micro_relat <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>((<span class="fu">tax_table</span>(floral_micro_relat)[, <span class="st">"Phylum"</span>] <span class="sc">%in%</span> top5phyla),</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                                 floral_micro_relat)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>floral_micro_relat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 456 taxa and 35 samples ]
sample_data() Sample Data:       [ 35 samples by 30 sample variables ]
tax_table()   Taxonomy Table:    [ 456 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<p>The final result yield us to a total of 456 ASV for downstream analysis.</p>
</section>
<section id="transforming-to-deseq2-object" class="level2">
<h2 class="anchored" data-anchor-id="transforming-to-deseq2-object">Transforming to Deseq2 object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(floral_micro_relat) <span class="ot">&lt;-</span> <span class="fu">meta</span>(floral_micro_relat) <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Block =</span> <span class="fu">factor</span>(Block, <span class="at">levels =</span> <span class="fu">seq</span>(<span class="dv">6</span>) ),</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">Time =</span> <span class="fu">factor</span>(Time), </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">Treatment =</span> <span class="fu">factor</span>(Treatment))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>diagdds_floral <span class="ot">&lt;-</span> <span class="fu">phyloseq_to_deseq2</span>(floral_micro_relat, </span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">design =</span>  <span class="sc">~</span>Block <span class="sc">+</span> Treatment <span class="sc">+</span> Time <span class="sc">+</span> Treatment<span class="sc">*</span>Time)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>gm_mean <span class="ot">=</span> <span class="cf">function</span>(x, <span class="at">na.rm=</span><span class="cn">TRUE</span>){</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">exp</span>(<span class="fu">sum</span>(<span class="fu">log</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>]), <span class="at">na.rm=</span>na.rm) <span class="sc">/</span> <span class="fu">length</span>(x))</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>geoMeans <span class="ot">=</span> <span class="fu">apply</span>(<span class="fu">counts</span>(diagdds_floral), <span class="dv">1</span>, gm_mean)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance estimation</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>diagdds_floral <span class="ot">=</span> <span class="fu">estimateSizeFactors</span>(diagdds_floral, <span class="at">geoMeans =</span> geoMeans)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>diagdds_floral <span class="ot">=</span> <span class="fu">estimateDispersions</span>(diagdds_floral, <span class="at">fitType=</span><span class="st">'local'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
</div>
<p>Extraction of the normalized matrix</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>diagvst_floral <span class="ot">=</span> <span class="fu">getVarianceStabilizedData</span>(diagdds_floral)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(diagvst_floral)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 456  35</code></pre>
</div>
</div>
<p>Now, we can replace the normalized matrix for the absolute counts in the phyloseq object</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>floral_micro_deseq_final <span class="ot">&lt;-</span> floral_micro_rel</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">otu_table</span>(floral_micro_deseq_final) <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(diagvst_floral, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-asv" class="level2">
<h2 class="anchored" data-anchor-id="differential-asv">Differential ASV</h2>
<p>Letâ€™s check first the design to make sure we have the right formula</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(diagdds_floral)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>~Block + Treatment + Time + Treatment * Time</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Defining alpha level</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>alpha <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Differential analysis</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>diagdds_floral <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(diagdds_floral, <span class="at">fitType =</span> <span class="st">"local"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>found already estimated dispersions, replacing these</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>results_deseq <span class="ot">&lt;-</span> <span class="fu">results</span>(diagdds_floral)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="co">#results_deseq &lt;- diagdds_floral[order(diagdds_floral$padj, na.last = NA), ]</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(diagdds_floral)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Intercept"           "Block_2_vs_1"        "Block_3_vs_1"       
 [4] "Block_4_vs_1"        "Block_5_vs_1"        "Block_6_vs_1"       
 [7] "Treatment_100_vs_25" "Treatment_200_vs_25" "Time_2_vs_1"        
[10] "Treatment100.Time2"  "Treatment200.Time2" </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>sig_dif_asv <span class="ot">&lt;-</span> results_deseq[<span class="fu">which</span>(results_deseq<span class="sc">$</span>padj <span class="sc">&lt;</span> alpha), ]</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>sig_dif_asv <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as</span>(sig_dif_asv,<span class="st">"data.frame"</span>),</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">as</span>(<span class="fu">tax_table</span>(floral_micro_deseq_final)[<span class="fu">rownames</span>(sig_dif_asv), ],</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"matrix"</span>))</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>sig_dif_asv <span class="ot">&lt;-</span> <span class="fu">remove_rownames</span>(sig_dif_asv)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>sig_dif_asv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   baseMean log2FoldChange     lfcSE     stat       pvalue        padj Kingdom
1 774.01032      0.8497361 0.2195828 3.869775 1.089359e-04 0.013827474   Bacte
2 647.18177      1.6183487 0.4256822 3.801777 1.436621e-04 0.013827474   Bacte
3 150.87652      3.3018801 0.7720993 4.276497 1.898573e-05 0.005427423   Bacte
4  61.03672      5.4152956 1.5311288 3.536800 4.050069e-04 0.025987945   Bacte
5  77.21488      9.1970051 2.1962575 4.187580 2.819441e-05 0.005427423   Bacte
6  24.81538      8.3578666 2.3324871 3.583242 3.393554e-04 0.025987945   Bacte
          Phylum            Class           Order           Family
1    Proteobacte Alphaproteobacte        Rhizobia        Rhizobiac
2 Verrucomicrobi    Verrucomicrob Verrucomicrobia      Rubritaleac
3    Proteobacte Alphaproteobacte        Rhizobia        Rhizobiac
4      Bacteroid         Bacteroi Sphingobacteria Sphingobacteriac
5    Proteobacte Gammaproteobacte    Burkholderia      Comamonadac
6    Proteobacte Gammaproteobacte    Burkholderia    Oxalobacterac
           Genus      Species
1  Pseudaminobac Unclassified
2     Luteolibac Unclassified
3          Shine           fu
4        Pedobac Unclassified
5       Rhizobac Unclassified
6 Janthinobacter Unclassified</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>